### ISSUE TITLE
Many objects remain undefined

### ISSUE THREAD

**Issue by XenoMustache** [fynnfluegge/oreon-engine#49]
Upon cloning the repository and building the maven projects in many different ways, including the way described by the "Getting Started" section in your Wiki, the project returns of 1k errors, stating many methods and objects remain undefined in most of the projects. I have checked that Eclipse has all of the dependencies it needs, and have even built the project using other tools such as Vscode, and IntelliJ IDEA, only to receive the exact same result.

**Comment by XenoMustache** [comment]
My mistake, I did not install Lomdok properly onto my IDE's when building the project. Actually taking a look at the instructions helped me resolve that issue. Now the project appears to be working as intended. Apologies for wasting your time.

**Comment by viloskovic** [comment]
@fynnfluegge In fact, I feel like using Lombok in opensource applications is not such a good practice. 
That way we force people to use and install Lombok and there is a high risk of creating unnecessary code caused by Lombok.

What is your feeling about this matter?

**Comment by fynnfluegge** [comment]
Hm ya you're kinda right. I started using lombok when I implemented Vulkan and I had hard deadlines, lombok saved lot of time for me...
But however, I just like lombok and how the code seems cleaner and more compact. It ensures that the getter and setter methods are nothing more than getters and setters, I saw code where somehow magic happened in getter/setter methods, lombok prevents that.
The problem to force people install lombok is a problem, thats true... I think this it's a trade-off...
For opensource it might be better to not use lombok, it could be confusing for people don't know lombok. But get rid of lombok now is very tiring work...
